lumen_rt_full cannot be compiled because of garbage_collect with `&mut []`

From https://github.com/lumen/lumen/runs/992513415

```
Compiling lumen_rt_full v0.1.0 (/__w/lumen/lumen/runtimes/full)
error[E0277]: the trait bound `liblumen_alloc::erts::process::gc::rootset::RootSet: std::convert::From<&mut [_; 0]>` is not satisfied
   --> runtimes/full/src/scheduler.rs:141:67
    |
141 | ...                   match arc_process.garbage_collect(0, &mut []) {
    |                                         ^^^^^^^^^^^^^^^ the trait `std::convert::From<&mut [_; 0]>` is not implemented for `liblumen_alloc::erts::process::gc::rootset::RootSet`
    |
    = help: the following implementations were found:
              <liblumen_alloc::erts::process::gc::rootset::RootSet as std::convert::From<&mut [liblumen_alloc::erts::term::arch::arch_x86_64::RawTerm]>>
              <liblumen_alloc::erts::process::gc::rootset::RootSet as std::convert::From<std::vec::Vec<liblumen_alloc::erts::term::boxed::Boxed<liblumen_alloc::erts::term::arch::arch_x86_64::RawTerm>>>>
    = note: required because of the requirements on the impl of `std::convert::Into<liblumen_alloc::erts::process::gc::rootset::RootSet>` for `&mut [_; 0]`

error: aborting due to previous error
```

--- a/runtimes/full/src/scheduler.rs
+++ b/runtimes/full/src/scheduler.rs
@@ -10,6 +10,7 @@ use std::sync::Arc;
 use liblumen_core::locks::RwLock;
 
 use liblumen_alloc::erts::exception::SystemException;
+use liblumen_alloc::erts::process::gc::RootSet;
 use liblumen_alloc::erts::process::{Priority, Process, Status};
 pub use liblumen_alloc::erts::scheduler::{id, ID};
 use liblumen_alloc::erts::term::prelude::*;
@@ -138,7 +139,8 @@ impl SchedulerTrait for Scheduler {
                                     Status::SystemException(system_exception) => {
                                         match system_exception {
                                             SystemException::Alloc(_) => {
-                                                match arc_process.garbage_collect(0, &mut []) {
+                                                let root_set = RootSet::new(&mut []);
+                                                match arc_process.garbage_collect(0, root_set) {
                                                     Ok(reductions) => {
                                                         arc_process.total_reductions.fetch_add(
                                                             reductions.try_into().unwrap(),

